### PR TITLE
Implemented BlockChangeCause for Form, Grow, Ignite and Remove

### PR DESCRIPTION
**Pull request by dredhorse**
Form =  Spread_Random, Spread_From, Forming
Grow
Ignite = Lava, Flint_and_Steel, Spread,  Lightning, Fireball
Remove = Fade, Melt, Disappear, Decay

all of those causes for the BlockChangeEvent still need to be implemented.

Signed-off-by: Don Redhorse dredhorse@breiden.net

### CODE REVIEW

File: src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+/**
+ * Caused when a block is formed or spreads
+ * todo needs to be implemented
+ */
+public class BlockFormCause extends BlockChangeCause {
+	/**
+	 * The different causes why a Block is formed in the world.
+	 */
+	public enum FORM_CAUSE {
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: This should be on a new line.

File: src/main/java/org/spout/vanilla/event/cause/BlockIgniteCause.java [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+/**
+ * Caused when a block is ignited
+ * todo needs be implemented
+ */
+public class BlockIgniteCause extends BlockChangeCause {
+	/**
+	 * The different causes why a Block was ignited.
+	 */
+	public enum IGNITE_CAUSE {
+		/**
```
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/cause/BlockIgniteCause.java]: And here

File: src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java (hunk 1 above)
Comment (dredhorse) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: the { ?

Comment (windy1) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: No the first enum value. "FORM_CAUSE"

Comment (dredhorse) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: that IS the enum...

Comment (kitskub) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: W1ndwaker, that's not an enum value. Dredhorse: FormCause

File: src/main/java/org/spout/vanilla/event/cause/BlockIgniteCause.java (hunk 2 above)
Comment (kitskub) [inline src/main/java/org/spout/vanilla/event/cause/BlockIgniteCause.java]: Same as above

File: src/main/java/org/spout/vanilla/event/cause/BlockRemoveCause.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+/**
+ * Caused when a block fades, melts, disappears or decays
+ * todo needs to be implemented
+ */
+public class BlockRemoveCause extends BlockChangeCause {
+	/**
+	 * The different causes why a Block is removed from the world.
+	 */
+	public enum REMOVE_CAUSE {
```
Comment (kitskub) [inline src/main/java/org/spout/vanilla/event/cause/BlockRemoveCause.java]: Same

File: src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java (hunk 1 above)
Comment (windy1) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: Oh herp, I just woke up, but anyways, enum names should never be all caps, name them as you would classes.

Comment (dredhorse) [inline src/main/java/org/spout/vanilla/event/cause/BlockFormCause.java]: ok.. will do in the other too... good morning :-)

**Pull request merged**